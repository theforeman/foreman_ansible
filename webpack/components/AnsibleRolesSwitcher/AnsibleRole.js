import React from 'react';
import { ListView, Icon } from 'patternfly-react';
import classNames from 'classnames';

import './styles.scss';

const clickHandler = (onClick, role) => (event) => {
  event.preventDefault();
  onClick(role);
};


const actionButton = (icon, onClick, role) => (
  <button
    href='#'
    onClick={clickHandler(onClick, role)}
    className="role-add-remove-btn"
  >
    <Icon className='fa-2x' type='fa' name={icon} />
  </button>
);

const AnsibleRole = ({ role, icon, onClick }) => (
  <ListView.Item
    id={role.id}
    className={classNames('listViewItem--listItemVariants', { 'ansible-role-disabled': role.inherited })}
    heading={role.name}
    actions={role.inherited ? '' : actionButton(icon, onClick, role)}
    stacked
  />
);

export default AnsibleRole;
