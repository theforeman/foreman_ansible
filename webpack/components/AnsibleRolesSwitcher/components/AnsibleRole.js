import React from 'react';
import { ListView } from 'patternfly-react';
import classNames from 'classnames';

import AnsibleRoleActionButton from './AnsibleRoleActionButton';
import '../AnsibleRolesSwitcher.scss';

const AnsibleRole = ({ role, icon, onClick }) => (
  <ListView.Item
    id={role.id}
    className={classNames('listViewItem--listItemVariants', { 'ansible-role-disabled': role.inherited })}
    heading={role.name}
    actions={role.inherited ?
             '' :
             <AnsibleRoleActionButton icon={icon} onClick={onClick} role={role}/>}
    stacked
  />
);

export default AnsibleRole;
