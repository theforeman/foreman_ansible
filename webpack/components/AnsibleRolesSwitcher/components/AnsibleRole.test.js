import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import AnsibleRole from './AnsibleRole';

const noop = () => {};

const fixtures = {
  'should render a role to add': {
    role: { name: 'test.role', id: 5 },
    icon: 'fa fa-plus-circle',
    onClick: noop,
  },
  'should render a role to remove': {
    role: { name: 'test.role', id: 5 },
    icon: 'fa fa-minus-circle',
    onClick: noop,
  },
  'should render inherited role to remove': {
    role: { name: 'test.role', id: 5, inherited: true },
    icon: 'fa fa-minus-circle',
    onClick: noop,
  },
};

describe('AnsibleRole', () => testComponentSnapshotsWithFixtures(AnsibleRole, fixtures));
