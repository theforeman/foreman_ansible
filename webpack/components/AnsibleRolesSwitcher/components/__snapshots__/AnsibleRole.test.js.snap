// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnsibleRole should render a role to add 1`] = `
<ListViewItem
  actions={
    <AnsibleRoleActionButton
      icon="fa fa-plus-circle"
      onClick={[Function]}
      role={
        Object {
          "id": 5,
          "name": "test.role",
        }
      }
    />
  }
  additionalInfo={null}
  checkboxInput={null}
  className="listViewItem--listItemVariants"
  compoundExpand={false}
  compoundExpanded={false}
  description={null}
  heading="test.role"
  hideCloseIcon={false}
  id={5}
  initExpanded={false}
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
/>
`;

exports[`AnsibleRole should render a role to remove 1`] = `
<ListViewItem
  actions={
    <AnsibleRoleActionButton
      icon="fa fa-minus-circle"
      onClick={[Function]}
      role={
        Object {
          "id": 5,
          "name": "test.role",
        }
      }
    />
  }
  additionalInfo={null}
  checkboxInput={null}
  className="listViewItem--listItemVariants"
  compoundExpand={false}
  compoundExpanded={false}
  description={null}
  heading="test.role"
  hideCloseIcon={false}
  id={5}
  initExpanded={false}
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
/>
`;

exports[`AnsibleRole should render inherited role to remove 1`] = `
<ListViewItem
  actions=""
  additionalInfo={null}
  checkboxInput={null}
  className="listViewItem--listItemVariants ansible-role-disabled"
  compoundExpand={false}
  compoundExpanded={false}
  description={null}
  heading="test.role"
  hideCloseIcon={false}
  id={5}
  initExpanded={false}
  leftContent={null}
  onCloseCompoundExpand={[Function]}
  onExpand={[Function]}
  onExpandClose={[Function]}
  stacked={true}
/>
`;
